Misc things: New textures, new creatures and new gold stacks + related code.

Bertram25 opened the pull request:
@hwoarangmy : Could you review that one and merge if ok?

Thanks! :)

Commit:
Made use of const string& when possible in the GameMap class.

Commit:
Danimal's New dormitory texture + new wall texture. Also made the walls colorable.

As discussed with Danimal.

hwoarangmy commented on source/GameMap.cpp:
```diff
@@ -1662,7 +1662,7 @@ std::vector<Room*> GameMap::getReachableRooms(const std::vector<Room*> &vec,
     return returnVector;
 }
 
-Room* GameMap::getRoomByName(std::string& name)
```
Thanks for correcting all the const I forgot :)

Commit:
Added support for new Gold Stacks from Danimal.

I also made the goldfullness enum more relevant and commented their use.

I removed the olf gold stack mesh as they really ugly compared to the new ones,
and I replaced one bed being the chest by the stronger gold stacks.

This is looking shiny! :)

Commit:
Added new and shiny Troll and Rock Golems by Dm3D, prepared by Danimal. :)

Commit:
Centered the exit pop-up

Commit:
Fixed an unhandled render request type seen in logs.

The Room Object deletion request wasn't handled.
